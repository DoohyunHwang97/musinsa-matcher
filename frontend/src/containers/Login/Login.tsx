import Axios from "axios";
import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

const Login = () => {
  const [inform, setInform] = useState<any>({
    username: "",
    password: "",
  });
  const { username, password } = inform;
  const navigate = useNavigate();
  const onClickRegister = () => {
    navigate("/register");
  };
  const handleInform = (e: any) => {
    setInform((prevState: any) => {
      return { ...prevState, [e.target.name]: e.target.value };
    });
  };

  const onSubmit = (e: any) => {
    e.preventDefault();
    const user = {
      username: inform.username,
      password: inform.password,
    };
    Axios.post("http://127.0.0.1:8000/api/clothes/login/", user)
      .then((res: any) => {
        if (res) {
          localStorage.clear();
          localStorage.setItem("token", res.data.session_key);
          navigate("/main");
        } else {
          setInform({
            username: "",
            password: "",
          });
          localStorage.clear();
        }
      })
      .catch((err: any) => {
        console.clear();
        alert("wrong");
      });
  };

  return (
    <div>
      <h2>Login</h2>
      <p>
        ID:
        <input name="username" value={username} onChange={handleInform} />
      </p>

      <p>
        PWD:
        <input
          type="password"
          name="password"
          value={password}
          onChange={handleInform}
        />
      </p>
      <button onClick={onClickRegister}>Register</button>
      <button onClick={onSubmit}>Sign In</button>
    </div>
  );
};
export default Login;
