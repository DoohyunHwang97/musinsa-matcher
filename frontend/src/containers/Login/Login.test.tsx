import React from "react";
import renderer from "react-test-renderer";
import * as router from "react-router";
import { expect } from "@jest/globals";
import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Login from "./Login";
const navigate = jest.fn();
beforeEach(() => {
  jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
});

describe("Login", () => {
  let component: any = null;

  it("initial render", () => {
    component = renderer.create(<Login />);
  });
  it("render correct", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("login test", () => {
    const login: any = render(<Login />);
    const input = login.getByTestId("username");
    fireEvent.change(input, { target: { value: "a" } });
    expect(input.value).toBe("a");
  });
  it("click register", () => {
    const login: any = render(<Login />);
    const button = login.getByTestId("register");
    fireEvent.click(button);
    expect(navigate).toHaveBeenCalledWith("/register");
  });
  it("click signin - no data", () => {
    global.alert = jest.fn();
    const login: any = render(<Login />);
    const button = login.getByTestId("signin");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledTimes(1);
  });
});
