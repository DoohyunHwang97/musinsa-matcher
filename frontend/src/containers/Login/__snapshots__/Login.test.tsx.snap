// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render correct 1`] = `
<div>
  <h2>
    Login
  </h2>
  <p>
    ID:
    <input
      data-testid="username"
      name="username"
      onChange={[Function]}
      value=""
    />
  </p>
  <p>
    PWD:
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </p>
  <button
    data-testid="register"
    onClick={[Function]}
  >
    Register
  </button>
  <button
    onClick={[Function]}
  >
    Sign In
  </button>
</div>
`;
